airbyte-db/src/main/java/io/airbyte/db/JdbcUtils.java

```diff
@@ -0,0 +1,231 @@
+
+  /**
+   * Map records returned in a result set.
+   *
+   * @param resultSet the result set
+   * @param mapper function to make each record of the result set
+   * @param <T> type that each record will be mapped to
+   * @return stream of records that the result set is mapped to.
+   */
+  public static <T> Stream<T> mapResultSet(ResultSet resultSet, CheckedFunction<ResultSet, T, SQLException> mapper) {
+    return StreamSupport.stream(new Spliterators.AbstractSpliterator<T>(Long.MAX_VALUE, Spliterator.ORDERED) {
+
+      @Override
+      public boolean tryAdvance(Consumer<? super T> action) {
+        try {
+          if (!resultSet.next())
+            return false;
+          action.accept(mapper.apply(resultSet));
+          return true;
+        } catch (SQLException e) {
+          throw new RuntimeException(e);
+        }
+      }
+
+    }, false);
+  }
+
+  /**
+   * Collect each record of a ResultSet into a list of JsonNode.
+   *
+   * @param resultSet the result set
+   * @return List of JsonNode.
+   * @throws SQLException exceptions throws when parsing the ResultSet.
+   */
+  public static List<JsonNode> resultSetToJson(ResultSet resultSet) throws SQLException {
+    return mapResultSet(resultSet, JdbcUtils::getJsonForRow).collect(Collectors.toList());
+  }
+
+  /**
+   * Stream the each record of a ResultSet lazily.
+   *
+   * @param preparedStatement that needs to be executed lazily.
+   * @return Stream of JsonNode.
+   * @throws SQLException exceptions throws when parsing the ResultSet.
+   */
+  public static <T> Stream<T> fetchStream(PreparedStatement preparedStatement, CheckedFunction<ResultSet, T, SQLException> mapper)
+      throws SQLException {
+    return Stream.of(1).flatMap(i -> {
+      try {
+        return mapResultSet(preparedStatement.executeQuery(), mapper);
+      } catch (SQLException e) {
+        throw new RuntimeException(e);
+      }
+    });
+  }
+
+  /**
+   * Stream the each record of a ResultSet as JsonNode lazily.
+   *
+   * @param preparedStatement that needs to be executed lazily.
+   * @return Stream of JsonNode.
+   * @throws SQLException exceptions throws when parsing the ResultSet.
+   */
+  public static Stream<JsonNode> fetchStreamAsJson(PreparedStatement preparedStatement) throws SQLException {
+    return fetchStream(preparedStatement, JdbcUtils::getJsonForRow);
+  }
+
+  /**
+   * Iterator of each record of a ResultSet as JsonNode.
+   *
+   * @param resultSet the result set
+   * @return Stream of JsonNode.
+   * @throws SQLException exceptions throws when parsing the ResultSet.
+   */
+  public static Iterator<JsonNode> resultSetToJsonIterator(ResultSet resultSet) throws SQLException {
+    return mapResultSet(resultSet, JdbcUtils::getJsonForRow).iterator();
+  }
+
+  public static JsonNode getJsonForRow(ResultSet r) throws SQLException {
```

Comment:
does it need to be public?
`rowToJson`

Reply:
yeah. this one does. it gets used by the AbstractJdbcSource (it is the mapper that is passed in).